onnx/optimizer/passes/split.h

```diff
@@ -0,0 +1,176 @@
+#pragma once
+
+#include "onnx/optimizer/passes/optimize_pass.h"
+
+namespace onnx { namespace optimization {
+
+static const char* impure_operators[] = {
+  "RandomNormal",
+  "RandomNormalLike",
+  "RandomUniform",
+  "RandomUniformLike"
+};
+
+static bool is_pure_operator(Node * n) {
+  for (auto x : impure_operators) {
+    if (n->kind() == Symbol(x)) {
+      return false;
+    }
+  }
+  return true;
+}
+
+// Split the graph into 'init' and 'predict' nets. This is kind of
```

Comment:
we should probably rename 'nets'. And maybe later - rename them into more generic functionality of staging

Reply:
Will follow this issue here: https://github.com/onnx/onnx/issues/439